src/server.cpp

```diff
@@ -0,0 +1,120 @@
+#include "server.hpp"
+
+#include <boost/asio.hpp>
+#include <nlohmann/json.hpp>
+#include <optional>
+
+using boost::asio::ip::tcp;
+
+namespace meow {
+Server::Server()
+    : io_context(), acceptor(io_context, tcp::endpoint(tcp::v4(), 0)) {}
+
+Server::~Server() = default;
+
+[[nodiscard]] std::string Server::get_port() const {
+  return std::to_string(acceptor.local_endpoint().port());
+}
+
+[[nodiscard]] std::vector<std::size_t> Server::get_clients_id() const {
+  std::vector<std::size_t> clients_id;
+  clients_id.reserve(client_threads.size());
+  for (auto& info : players_info) {
+    clients_id.push_back(info.id);
+  }
+  return clients_id;
+}
+
+void Server::handle_client(tcp::socket& socket) {
+  tcp::iostream client(std::move(socket));
+  std::size_t client_id =
+      std::hash<std::thread::id>{}(std::this_thread::get_id());
```

Comment:
why do we need to compute hash for client ids? Why not make them consecutive?

Reply:
otherwise it need to lock mutex on almost whole part of handshake